motor/providers/microsoft/platform.go

```diff
@@ -0,0 +1,52 @@
+package microsoft
+
+import (
+	"github.com/cockroachdb/errors"
+	"go.mondoo.com/cnquery/motor/platform"
+	"go.mondoo.com/cnquery/motor/providers"
+	"go.mondoo.com/cnquery/motor/vault"
+)
+
+func (p *Provider) Identifier() (string, error) {
+	if p.assetType == azure {
+		return "//platformid.api.mondoo.app/runtime/azure/subscriptions/" + p.subscriptionID, nil
+	}
+	return "//platformid.api.mondoo.app/runtime/ms365/tenant/" + p.tenantID, nil
+}
+
+func (p *Provider) SubscriptionID() string {
+	return p.subscriptionID
+}
+
+func (p *Provider) TenantID() string {
+	return p.tenantID
+}
+
+func (p *Provider) ClientID() string {
+	return p.clientID
+}
+
+func (p *Provider) Credential() *vault.Credential {
+	return p.cred
+}
+
+func (p *Provider) PlatformInfo() (*platform.Platform, error) {
+	switch p.assetType {
+	case azure:
```

Comment:
I think this highlights that we want to defined azure ad as a separate asset going forward.

Reply:
`azure.ad` is marked as deprecated, as that's the same as `msgraph.beta.* We still need a subscription for any other azure resources though (vms, network interfaces, etc)